src/mixins/raster-layer.js

```diff
@@ -158,6 +161,34 @@ export default function rasterLayer (layerType) {
     }
   }
 
+  _layer.getColorDomain = function (chart) {
+    const subquery = _layer.genSQL({
+      table: _layer.crossfilter().getTable()[0],
+      width: Math.round(chart.width() * chart._getPixelRatio()),
+      height: Math.round(chart.height() * chart._getPixelRatio()),
+      min: chart.conv4326To900913(chart._minCoord),
+      max: chart.conv4326To900913(chart._maxCoord),
+      filter: _layer.crossfilter().getFilterString()
+    })
+
+    const sql = `SELECT MIN(c.color) as minimum, MAX(c.color) as maximum, STDDEV(c.color) as deviation, AVG(c.color) as mean FROM (${subquery}) as c`
```

Comment:
yes, it's a bit strange right now. when we start in implementing multi-layer in immerse, i plan to tackle and clean up these architectural issues